blockscout-ens/bens-logic/src/subgraphs_reader/sql/domain.rs

```diff
@@ -261,35 +271,36 @@ pub async fn batch_search_addresses(
         ORDER BY resolved_address, created_at
         "#,
     ))
-    .bind(addresses)
+    .bind(bind_string_list(addresses))
     .fetch_all(pool)
     .await?;
 
     Ok(domains)
 }
 
-// TODO: rewrite to sea_query generation
 #[instrument(
-    name = "batch_search_addresses_cached",
-    skip(pool, addresses),
-    fields(job_size = addresses.len()),
+    name = "batch_search_addr_reverse_names",
+    skip(pool, addr_reverse_hashes),
+    fields(job_size = addr_reverse_hashes.len()),
     err(level = "error"),
     level = "info",
 )]
-pub async fn batch_search_addresses_cached(
+pub async fn batch_search_addr_reverse_names(
     pool: &PgPool,
     schema: &str,
-    addresses: &[&str],
-) -> Result<Vec<DomainWithAddress>, SubgraphReadError> {
-    let domains: Vec<DomainWithAddress> = sqlx::query_as(&format!(
+    addr_reverse_hashes: &[impl AsRef<str>],
+) -> Result<Vec<ReverseRecord>, SubgraphReadError> {
+    let domains: Vec<ReverseRecord> = sqlx::query_as(&format!(
         r#"
-        SELECT id, domain_name, resolved_address
-        FROM {schema}.address_names
-        where
-            resolved_address = ANY($1)
+        SELECT d.id as addr_reverse_id, nc.name as reversed_name
+        FROM {schema}.domain d
+        JOIN {schema}.name_changed nc ON nc.resolver = d.resolver
+        WHERE d.id = ANY($1)
+            AND d.block_range @> 2147483647
+        ORDER BY nc.block_number DESC;
```

Comment:
I'm don't quite understand why the direct query is simpler than for the cached view?

Reply:
because in direct approach we have to do a lot 2 SQL queries, see `resolve_addresses::resolve_addr_reverse` vs `resolve_addresses::resolve_addr_reverse_cached`